src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Withdraw/components/Confirm.tsx

```diff
@@ -231,6 +231,7 @@ export const Confirm: React.FC<ConfirmProps> = ({ accountId, onNext }) => {
 
     try {
       const position = await getThorchainSaversPosition({ accountId, assetId })
+      if (!position) throw new Error(`No position found for assetId: ${assetId}`)
```

Comment:
don't love that this throws, catches, rethrows, then is setting a monad. maybe return empty string and have the consumer do `maybeUtxoAccountAddress && setUtxoAccountAddress(maybe...)`?

also, given `getMaybeFromUtxoAccountAddress` is only used in one spot, could consolidate the callback and effect?

Reply:
Definitely, makes things a lot cleaner: 27ce5d7af10b8cddb855041320222dc19cb0c45e